vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -1,1017 +1,1023 @@
+               array.splice(index, 1);
+             }
+          }
         }
-      }
-      // Let server know we're done
-      grid.$server.confirmParentUpdate(id, parentKey);
-    };
-
-    grid.$connector.confirm = function(id) {
-      // We're done applying changes from this batch, resolve outstanding
-      // callbacks
-      let outstandingRequests = Object.getOwnPropertyNames(rootPageCallbacks);
-      for(let i = 0; i < outstandingRequests.length; i++) {
-        let page = outstandingRequests[i];
-        let lastRequestedRange = lastRequestedRanges[root] || [0, 0];
-        // Resolve if we have data or if we don't expect to get data
-        if ((cache[root] && cache[root][page]) || page < lastRequestedRange[0] || page > lastRequestedRange[1]) {
-          let callback = rootPageCallbacks[page];
-          delete rootPageCallbacks[page];
-          callback(cache[root][page] || new Array(grid.pageSize));
-          // Makes sure to push all new rows before this stack execution is done so any timeout expiration called after will be applied on a fully updated grid
-          //Resolves https://github.com/vaadin/vaadin-grid-flow/issues/511
-          if(grid._debounceIncreasePool){
-              grid._debounceIncreasePool.flush();
+      })
+
+      grid.$connector.confirmParent = tryCatchWrapper(function(id, parentKey, levelSize) {
+        if(!treePageCallbacks[parentKey]) {
+          return;
+        }
+        if(cache[parentKey]) {
+          cache[parentKey].size = levelSize;
+        }
+        let outstandingRequests = Object.getOwnPropertyNames(treePageCallbacks[parentKey]);
+        for(let i = 0; i < outstandingRequests.length; i++) {
+          let page = outstandingRequests[i];
+
+          let lastRequestedRange = lastRequestedRanges[parentKey] || [0, 0];
+          if((cache[parentKey] && cache[parentKey][page]) || page < lastRequestedRange[0] || page > lastRequestedRange[1]) {
+            let callback = treePageCallbacks[parentKey][page];
+            delete treePageCallbacks[parentKey][page];
+            let items = cache[parentKey][page] || new Array(levelSize);
+            callback(items, levelSize);
           }
+        }
+        // Let server know we're done
+        grid.$server.confirmParentUpdate(id, parentKey);
+      });
 
+      grid.$connector.confirm = tryCatchWrapper(function(id) {
+        // We're done applying changes from this batch, resolve outstanding
+        // callbacks
+        let outstandingRequests = Object.getOwnPropertyNames(rootPageCallbacks);
+        for(let i = 0; i < outstandingRequests.length; i++) {
+          let page = outstandingRequests[i];
+          let lastRequestedRange = lastRequestedRanges[root] || [0, 0];
+          // Resolve if we have data or if we don't expect to get data
+          if ((cache[root] && cache[root][page]) || page < lastRequestedRange[0] || page > lastRequestedRange[1]) {
+            let callback = rootPageCallbacks[page];
+            de
```

Comment:
This seems to only be called from within other methods that are already using `tryCatchWrapper` so it could be removed here.

Reply:
Removed.